### PR TITLE
Chats redesign

### PR DESCRIPTION
**Pull request by Aldemylla**
Goals:
  - Improve reading and viewing of messages
  - Improve the signaling of events in chat
  - Adapt the layout to a grid that works perfectly
  - Improve the visualization of contacts in the queue sidebar
 
  Improvements:
- Improved User Interface: We have completely overhauled the chat interface, making it more intuitive, attractive and easier to use. This includes improving the layout of elements, using more consistent colors and icons, and improving overall navigability.
- Bug Fixes: Several bugs and issues reported by users have been resolved. This includes fixes for undelivered messages, formatting issues, and other bugs that affected the module's functionality.
- Responsiveness: Chats components are now fully responsive, ensuring a consistent experience on mobile devices and desktops.

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -65,10 +66,17 @@ export default {
...
+      appProject: (state) => state.config.project,
     }),
 
     configsForInitializeWebSocket() {
-      return [this.$store.state.config.token, this.$store.state.config.project];
+      const { appToken, appProject } = this;
+
+      return [appToken, appProject];
+    },
+    teste() {
```
Comment (paulobernardoaf) [inline src/App.vue:78]: Should this be here? 👀

File: src/components/chats/ContactInfo/index.vue [hunk 2]
```diff
@@ -1,227 +1,225 @@
...
-                @update-current-custom-field="updateCurrentCustomField"
-                @save-value="saveCurrentCustomFieldValue"
-              />
-            </template>
+              }}</span>
+            </div>
+          </section>
+        </aside-slot-template-section>
+
+        <!-- <aside-slot-template-section v-if="isHistory">
```
Comment (paulobernardoaf) [inline src/components/chats/ContactInfo/index.vue:103]: Dead code?

File: src/components/chats/ContactInfo/index.vue [hunk 3]
```diff
@@ -1,227 +1,225 @@
...
-            <p class="username">
-              {{ contact.name }}
-            </p>
-            <template>
-              <p style="margin-bottom: 0.75rem">
-                <span class="title"> {{ contactNumberClosedChat.plataform }}: </span>
-                {{ contactNumberClosedChat.contactNum }}
-              </p>
-            </template>
+      <!-- <section v-if="isHistory" class="scrollable">
```
Comment (paulobernardoaf) [inline src/components/chats/ContactInfo/index.vue:149]: Dead code?

File: src/components/chats/chat/ChatMessages.vue [hunk 4]
```diff
@@ -1,93 +1,119 @@
...
-        <span class="chat-messages__room__divisor__label">{{ $t('chat_closed_by.agent') }}</span>
-        <div class="chat-messages__room__divisor__line" />
-      </section>
-      <section>
-        <div v-if="room.is_waiting" class="chat-messages__room__transfer-info">
-          {{ $t('waiting_answer.waiting_cliente_answer') }}
-        </div>
-      </section>
+    <!-- Closed chat tags  -->
+    <!-- <chat-feedback
```
Comment (paulobernardoaf) [inline src/components/chats/chat/ChatMessages.vue:105]: Dead code?

**Review by luishsf — APPROVED** [review]

File: src/components/chats/chat/ChatMessages.vue [hunk 5]
```diff
@@ -152,19 +184,34 @@ export default {
     prevRoomUuid: null,
   }),
 
+  mounted() {
+    window.addEventListener('online', () => {
+      this.resendMessages();
+    });
+
+    // const observer = new IntersectionObserver((entries) => {
```
Comment (paulobernardoaf) [inline src/components/chats/chat/ChatMessages.vue:192]: Dead code?

File: src/components/chats/chat/ChatMessages.vue [hunk 6]
```diff
@@ -152,19 +184,34 @@ export default {
...
-    isHistory() {
-      return !this.room.is_active;
-    },
+    ...mapState({
+      roomMessagesSendingUuids: (state) => state.roomMessages.roomMessagesSendingUuids,
+      roomMessagesFailedUuids: (state) => state.roomMessages.roomMessagesFailedUuids,
+      roomMessages: (state) => state.roomMessages.roomMessages,
+      roomMessagesSorted: (state) => state.roomMessages.roomMessagesSorted,
+    }),
+    // rooms() {
```
Comment (paulobernardoaf) [inline src/components/chats/chat/ChatMessages.vue:209]: Dead code?

File: src/components/chats/chat/ChatMessages.vue [hunk 7]
```diff
@@ -211,36 +412,50 @@ export default {
...
       if (imageIndex - 1 >= 0) {
         this.currentMedia = this.medias[imageIndex - 1];
       }
     },
 
+    handleScroll() {
+      const { chatMessages } = this.$refs;
+      if (!chatMessages) return;
+
+      // if (this.isEdgeRoomMessage(this.roomMessagesSorted[0].minutes)) {
```
Comment (paulobernardoaf) [inline src/components/chats/chat/ChatMessages.vue:432]: Dead code?

File: src/layouts/ChatsLayout/components/TheRoomList/RoomGroup/ContactRoom.vue [hunk 8]
```diff
@@ -93,6 +34,7 @@ export default {
       this.waitingTime = this.room.waitingTime;
       this.timer = setInterval(() => {
         this.waitingTime += 1;
+        console.log(1);
```
Comment (paulobernardoaf) [inline src/layouts/ChatsLayout/components/TheRoomList/RoomGroup/ContactRoom.vue:37]: remove console

File: src/services/api/resources/chats/history.js [hunk 9]
```diff
@@ -0,0 +1,76 @@
+import http from '@/services/api/http';
+import { getProject } from '@/utils/config';
+
+export default {
+  // async getAllWithClosedRooms(offset, limit, contact, date, tags, sector) {
```
Comment (paulobernardoaf) [inline src/services/api/resources/chats/history.js:5]: dead code?

File: src/services/api/resources/chats/message.js [hunk 10]
```diff
@@ -53,26 +69,12 @@ export default {
       },
       {
         onUploadProgress: (event) => {
+          // const progress = parseInt(Math.round((event.loaded * 100) / event.total), 10);
```
Comment (paulobernardoaf) [inline src/services/api/resources/chats/message.js:72]: dead code?

File: src/views/chats/ClosedChats/index.vue [hunk 11]
```diff
@@ -0,0 +1,213 @@
...
+      const project = await ProjectApi.getInfo();
+      this.project = project.data;
+      this.isLoadingHeader = false;
+    },
+
+    async chatScrollTop() {
+      if (this.roomMessagesNext) {
+        this.getHistoryContactRoomMessages();
+      } else {
+        // const roomUuidIndex = this.selectedRoomsUuids?.findIndex(
```
Comment (paulobernardoaf) [inline src/views/chats/ClosedChats/index.vue:124]: dead code?

**Review by paulobernardoaf — APPROVED** [review]

File: src/views/chats/ClosedChats/index.vue (hunk 11 above)
Comment (Aldemylla) [reply src/views/chats/ClosedChats/index.vue:124]: No, this code is commented to handle future updates with the backend

File: src/components/chats/chat/ChatMessages.vue (hunk 5 above)
Comment (Aldemylla) [reply src/components/chats/chat/ChatMessages.vue:192]: No, this code is commented to handle future updates with the backend

File: src/components/chats/chat/ChatMessages.vue (hunk 4 above)
Comment (Aldemylla) [reply src/components/chats/chat/ChatMessages.vue:105]: No, this code is commented to handle future updates with the backend

File: src/components/chats/chat/ChatMessages.vue (hunk 7 above)
Comment (Aldemylla) [reply src/components/chats/chat/ChatMessages.vue:432]: No, this code is commented to handle future updates with the backend

File: src/components/chats/ContactInfo/index.vue (hunk 2 above)
Comment (Aldemylla) [reply src/components/chats/ContactInfo/index.vue:103]: No, this code is commented to handle future feature

**Pull request merged**